src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/redundantthis/InputRedundantThisCatchVariables.java

```diff
@@ -0,0 +1,39 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.redundantthis;
+
+public class InputRedundantThisCatchVariables {
+    private Throwable ex;
+
+    public InputRedundantThisCatchVariables(Throwable ex) {
+        this.ex = ex; // no violation
+    }
+
+    public void run() {
+        if (this.ex != null) { // violation
+            try {
+                exceptional(this.ex); // violation
+            }
+            catch (RuntimeException ex) {
+                if (ex == this.ex) { // no violation
+                    debug("Expected exception thrown", ex);
+                }
+                else {
+                    ex.printStackTrace();
+                }
+            }
+            catch (Error err) {
+                if (err == this.ex) { // violation
+                    this.debug("Expected exception thrown", err); // violation
+                }
+                else {
+                    ex.printStackTrace();
+                }
```

Comment:
again about redundant things - else block here and above is not needed,

Reply:
Done.